add board preview, expandable pocket component

merowin opened the pull request:
issue: #187 

Adds a board preview component, alongside a component "expandable pocket" which can be used to expand / collapse arbitrary content.

![grafik](https://github.com/user-attachments/assets/3fca4458-b3d6-4220-a9f9-c8442757baf6)

merowin commented on packages/vue-client/src/components/boards/BoardPreview.vue:
```diff
@@ -0,0 +1,117 @@
+<script setup lang="ts">
```
There is some code duplicated from other board components here.

benjaminpjones replied:
Wait, why copy the code?  This feels like something we can just use [`DefaultBoard`](https://github.com/govariantsteam/govariants/blob/020709a8145b2fa7c66e5327b1379433cf0de587/packages/vue-client/src/components/boards/DefaultBoard.vue) for

merowin replied:
Hm good point, we could wrap it with `pointer-events: none` to prevent the hover effects, and get star points etc. for free.

merowin commented:
Ok, so I deleted the board preview component, and use the DefaultBoard instead. I made the game state prop optional, so we don't need to create a dummy state for the preview.
Also I changed the expandable pocket component so it looks nicer, including an expand / collapse animation.

![grafik](https://github.com/user-attachments/assets/fcf9cf7d-9057-4372-b2ce-1f7625d0919a)
